synapse/handlers/federation_event.py

```diff
@@ -862,6 +862,10 @@ async def _process_pulled_event(
             self._sanity_check_event(event)
         except SynapseError as err:
             logger.warning("Event %s failed sanity check: %s", event_id, err)
+            if backfilled:
```

Comment:
why do we only do this if `backfilled` is true? Doesn't the same apply however we ended up pulling this event?

Reply:
Talking it out further in https://github.com/matrix-org/synapse/pull/13589#discussion_r961064218, it does make sense to drop the condition.

Whether that be this PR or a future PR though for generally tracking wherever we fail to fetch or process a federated event.

Reply:
One reason it would be worth deciding this now is that it will affect the name of the table: is it "failed to process backfilled event" or "failed to process pulled event" ?

Reply:
"failed to process pulled event" makes sense and is probably better.

But I think it also needs to track "failed to fetch event" for the case where `/backfill` just doesn't include the event we asked about or if `/event` throws a 404, it would be good to also track that.

`event_failed_pull_attempts` seems to encompass all of these conditions ⏩ wdyt?

Reply:
ugh gosh, those sound like quite different things to be honest. I'm not sure we should conflate "`/backfill` just doesn't include the event we asked about" with a failure to fetch a copy of an event via `/event`? To me the former doesn't necessarily imply we shouldn't try to fetch the event via `/event`.

Honestly... I think we need to think harder about the interactions here, and I'm afraid I can't commit to doing so before I go on leave tomorrow. You might be able to get some engagement from someone else in the Synapse team.

Reply:
👍 on getting the right database name in this PR 

Created https://github.com/matrix-org/synapse/issues/13700 to track the follow-up for generalizing the failed to fetch/process of federated events though.

Reply:
> ugh gosh, those sound like quite different things to be honest. I'm not sure we should conflate "`/backfill` just doesn't include the event we asked about" with a failure to fetch a copy of an event via `/event`? To me the former doesn't necessarily imply we shouldn't try to fetch the event via `/event`.

I could see that. I've updated this one as a maybe in https://github.com/matrix-org/synapse/issues/13700

---

Will get some more thoughts 👍

Reply:
I'm afraid that this is over my head — I don't know enough about the various ways of fetching events to tell you what the repercussions of conflating these are; it is probably worth fetching someone who does know to have a look. I'll put this back on the general review queue and I'll make a noise about it in synapse-dev, but you may have to chase someone up when you're back online.

Reply:
`event_failed_pull_attempts` sounds good ⏩ 

Discussed with @erikjohnston; We can draw the line at and not include when we ask `/backfill` from a certain event but the `/backfill` response doesn't include it. We include errors from attempts to fetch the event, ignoring any backfill specific logic.

This could possibly be a different table to track dead extremities. But we can think about whether we actually need that later.